modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/PartitionScanPublisher.java

```diff
@@ -233,8 +233,9 @@ private CompletableFuture<Void> retrieveAndProcessBatch() {
         }
 
         private void processBatch(Collection<T> batch, int requestedCnt) {
-            assert batch != null;
-            assert batch.size() <= requestedCnt : "Rows more than requested " + batch.size() + " " + requestedCnt;
+            assert batch != null : "Batch is null";
```

Comment:
This message is not needed in general, but if you think it is necessary, I will not force it.

Reply:
If an assert without a message gets triggered, it produces an AssertionError in which getMessage() returns null. If then this error gets wrapped in IgniteInternalException, it will also contain null as getMessage(). As a result, the final exception will also have no message at all, so it might be really difficult to find the root cause in logs.